aws_lambda/aws_lambda.py

```diff
@@ -432,6 +453,36 @@ def update_function(cfg, path_to_zip_file):
 
     client.update_function_configuration(**kwargs)
 
+def upload_s3(cfg, path_to_zip_file):
+    """Upload a function to AWS S3."""
+
+    print('Uploading your new Lambda function')
+    byte_stream = read(path_to_zip_file, binary_file=True)
+    aws_access_key_id = cfg.get('aws_access_key_id')
+    aws_secret_access_key = cfg.get('aws_secret_access_key')
+
+    account_id = get_account_id(aws_access_key_id, aws_secret_access_key)
+    role = get_role_name(account_id, cfg.get('role', 'lambda_basic_execution'))
```

Comment:
is it safe to assume that the role for execution will be the same one used for uploading bundles to s3? Maybe this should be configured independently. 

Reply:
Yeah, I agree. I tweaked this role when I was testing but it really should be something like `basic_s3_upload`.